backend/dataall/core/environment/services/environment_service.py

```diff
@@ -868,13 +868,19 @@ def delete_environment(uri):
                     f'related objects before proceeding',
                 )
             else:
-                PolicyManager(
-                    role_name=environment.EnvironmentDefaultIAMRoleName,
-                    environmentUri=environment.environmentUri,
-                    account=environment.AwsAccountId,
-                    region=environment.region,
-                    resource_prefix=environment.resourcePrefix,
-                ).delete_all_policies()
+                if StackRepository.find_stack_by_target_uri(session, environment.environmentUri) not in [
```

Comment:
In case auto pivotRole creation the pivot role won't exist as well (I've come across this issue a few times) so I think that this check will fail regardless. Wdyt?